### PR TITLE
Node 4.x and reduced dependencies

### PR DESCRIPTION
**Pull request by thom4parisot**
For some reasons the `build:source` command does not work (it displays the help prompt) but you get the spirit.

Plus I simplified your Dockerfiles to have only one, called with different (and explicit) commands.

I removed phantomjs from the test Docker image but it could certainly be added as a devDependencies.

Also builds have to be released with:

``` bash
npm version patch
npm publish
```

Or any other level like `minor` or `major`. This way you always ensure you publish a version which has been tested. And you can automated the npm publish via Travis, if a `$GIT_TAG` is available in the env.

### CODE REVIEW

**Comment by vallettea** [comment]
:+1:

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  volumes:
+      - .:/usr/6bin
+  tty: true
+
+tests:
+  <<: *server
+  command: npm test
+  links:
+      - server
+  ports:
```
Comment (vallettea) [inline docker-compose.yml:16]: that's cool

**Comment by vallettea** [comment]
@DavidBruant I let you merge it

**Comment by bleucitron** [comment]
Thx @oncletom for this PR.
It is a great example of how things can be done better with Dockerfiles and npm scripts.

However, the project's context makes part of your changes irrelevant:
- When we tried to put Docker on our sensors, we had some issues with Node 4, because a pair of strategic dependencies were not compatible, so we sticked with Node 0.10
- Docker use for our sensors (and thus this 6bin project) was something we tried at first, but for hardware reasons we chose to drop it.

Sorry, we should have totally cleaned up the repo of all Docker refs :/

The `npm-run-all` stuff looks really interesting though, and when i have some time, i'll look into it to make it work as I like.

**Pull request closed without merging**